src/Infrastructure/Identity/TokenService.cs

```diff
@@ -141,15 +141,10 @@ private string GenerateEncryptedToken(SigningCredentials signingCredentials, IEn
 
     private ClaimsPrincipal GetPrincipalFromExpiredToken(string token)
     {
-        if (string.IsNullOrEmpty(_jwtSettings.Key))
-        {
-            throw new InvalidOperationException("No Key defined in JwtSettings config.");
-        }
-
         var tokenValidationParameters = new TokenValidationParameters
         {
             ValidateIssuerSigningKey = true,
-            IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtSettings.Key)),
+            IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtSettings.Key!)),
```

Comment:
Couldn't we now make the Key property actually non-nullable with the validation in place? So we don't even have to use the bang (!) everywhere?

Reply:
Same for the DBProvider and ConnectionString properties of the databasesettings options?

Reply:
If we make them non-nullable, there will need to be default values. `string.Empty`? The string.Empty will still be caught by validators, but still ensure we aren't forcing a null into a supposedly non-nullable property.

Reply:
I don't think it's a problem initializing with `= default!` here... as the validator should catch that anyways.